File: DoAnChuyenNganh-main/DoAnCoSo/WebDuLichDaLat/Models/BlogPost.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace WebDuLichDaLat.Models
{
    public class BlogPost
    {
        public int Id { get; set; }

        [Required, StringLength(150)]
        public string Title { get; set; }

        [Required]
        public string Content { get; set; }

        public string? ImageUrl { get; set; }

        [DataType(DataType.Date)]
        public DateTime PostedDate { get; set; } = DateTime.Now;

        public string? Author { get; set; }
    }
}

File: src/main/java/com/example/dalattravel/model/BlogPost.java
package com.example.DaLattravel.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "blog_posts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BlogPost {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    private String imageUrl;

    @Builder.Default
    private LocalDateTime postedDate = LocalDateTime.now();

    private String author;
}
